src/main.rs

```diff
@@ -3,13 +3,21 @@ use std::{
     fs,
     io::{stdin, Read},
     path::{Path, PathBuf},
+    sync::mpsc::{channel, Receiver, Sender},
```

Comment:
Note that `std::sync::mpsc` module is a pretty poor implementation of channels, and shouldn't be used for anything serious. The crate of choice instead is `crossbeam_channel`. However, we don't use channels for anything important, so sticking with std seems fine.

Reply:
Hm, actually, as `ignore` itself depends on crossbeam_channel, it's better to use it instead

Reply:
ok that's good to know. I guess that crossbeam_channel would also blow-up the number of dependencies and binary size.

Reply:
oops I missed your second comment. I will look into it then.

Reply:
It wouldn't blow the number of dependencies, as we already transitively depend on it. 

As for the size of the binary, using both impls would blow the size up, but using only one should be an improvement, if the impls have some monophonic parts (and I think they might have them?). 

In general, just adding dependency won't increase the binary size, because unused code from stdlib is stripped anyway, and, if the code in question is generic, it's not included in the first place, as there are no monomorphizations.

Reply:
ok that was dead easy: 2dd75bd428b148b6dd9857f8c87ffc1a360e03b2